netmaster/docknet/docknet.go

```diff
@@ -160,23 +164,24 @@ func CreateDockNet(tenantName, networkName, serviceName string, nwCfg *mastercfg
 		ipamOptions["tenant"] = nwCfg.Tenant
 		ipamOptions["network"] = nwCfg.NetworkName
 
+		ipamCfg := network.IPAM{
+			Driver:  ipamDriverName,
+			Config:  ipams,
+			Options: ipamOptions,
+		}
 		// Build network parameters
-		nwCreate := dockerclient.NetworkCreate{
-			Name:           docknetName,
+		nwCreate := types.NetworkCreate{
 			CheckDuplicate: true,
 			Driver:         netDriverName,
-			IPAM: dockerclient.IPAM{
-				Driver:  ipamDriverName,
-				Config:  ipams,
-				Options: ipamOptions,
-			},
-			Options: netPluginOptions,
+			IPAM:           &ipamCfg,
+			Options:        netPluginOptions,
+			Attachable:     true,
 		}
 
 		log.Infof("Creating docker network: %+v", nwCreate)
 
 		// Create network
-		resp, err := docker.CreateNetwork(&nwCreate)
+		resp, err := docker.NetworkCreate(context.Background(), docknetName, nwCreate)
```

Comment:
docker client library uses cancellable threads for the REST access and hence require a context to be passed. Here we are using the Background context which doesn't have a timeout or deadline. It waits until it gets a response or http error and the NetworkCreate api returns the response/error.